src/core/DataSourceBase.test.ts

```diff
@@ -163,6 +170,199 @@ describe('DataSourceBase', () => {
+            mockGet$.mockReturnValueOnce(of('observable-test'));
+
+            const response = await firstValueFrom(dataSource.get$('/test-endpoint', { param1: 'value1' }));
+
+            expect(mockGet$).toHaveBeenCalledWith(
+                backendSrv,
+                '/test-endpoint',
+                { param1: 'value1' }
+            );
+            expect(response).toEqual('observable-test');
+        });
+
+        it('should send GET$ request with API ingress when useApiIngress is true', async () => {
+            mockGet$.mockReturnValueOnce(of('observable-test'));
+
+            const response$ = dataSource.get$('/test-endpoint', { param1: 'value1' }, true);
+            const response = await firstValueFrom(response$);
+
+            expect(mockApiSessionUtils.createApiSession).toHaveBeenCalled();
+            expect(mockGet$).toHaveBeenCalledWith(
+                backendSrv,
+                "http://api-ingress.com/test-endpoint",
+                {
+                    'param1': 'value1',
+                    'x-ni-api-key': 'api-key-secret'
+                }
+            );
+            expect(response).toEqual('observable-test');
+        });
+
+        it('should not send GET$ request if no API session is returned', async () => {
+            jest.clearAllMocks();
+            mockApiSessionUtils.createApiSession.mockRejectedValueOnce(new Error('No session created'));
+
+            const response$ = dataSource.get$('/test-endpoint', { param1: 'value1' }, true);
+
+            await expect(firstValueFrom(response$)).rejects.toThrow('No session created');
+            expect(mockApiSessionUtils.createApiSession).toHaveBeenCalled();
+            expect(mockGet$).not.toHaveBeenCalled();
+        });
+    });
+
+    describe('post$', () => {
+        it('should send POST$ request with correct parameters when useApiIngress is not set', async () => {
+            mockPost$.mockReturnValueOnce(of('observable-test'));
+
+            const response = await firstValueFrom(dataSource.post$('/test-endpoint', { body: 'body' }, { options: 'optionValue' }));
+
+            expect(mockPost$).toHaveBeenCalledWith(
+                backendSrv,
+                '/test-endpoint',
+                { body: 'body' },
+                { options: 'optionValue' }
+            );
+            expect(response).toEqual('observable-test');
+        });
+
+        it('should send POST$ request with API ingress when useApiIngress is true', async () => {
+            mockPost$.mockReturnValueOnce(of('observable-test'));
+
+            const response$ = dataSource.post$(
+                '/test-endpoint',
+                { body: 'body' },
+                {
+                    headers: {
+                        testHeader: 'headerValue'
+                    }
+                },
+                true
+            );
+            const response = await firstValueFrom(response$);
```

Comment:
Let's follow consistent pattern.


```suggestion
            const response = await firstValueFrom(dataSource.post$(
                '/test-endpoint',
                { body: 'body' },
                {
                    headers: {
                        testHeader: 'headerValue'
                    }
                },
                true
            ));
```